client/src/types.ts

```diff
@@ -204,5 +204,5 @@ export interface IAuditSettings {
   online: boolean | null
   randomSeed: string | null
   riskLimit: number | null
-  auditType: 'BALLOT_POLLING' | 'BATCH_COMPARISON'
+  auditType: 'BALLOT_POLLING' | 'BATCH_COMPARISON' | null
```

Comment:
I actually don't think this can ever be null, since it gets set when the audit is created. it's not allowed to be null in the db

Reply:
@jonahkagan This is why I needed to update the `selectedValue={values.auditType || 'BALLOT_POLLING'}` bit. Defaulting to 'BALLOT_POLLING' was breaking the conditions since you only care about waiting for the batch tallies on batch audits. And before the apis were returning, it was causing lots of issues. So I defaulted the setting to null like the rest of the settings, and just set the initial field to default to 'BALLOT_POLLING' if the value was null. But of course that needed to add null to the type like the rest. So basically it's only null before it hears back from the api.

Reply:
hmm, what if the entire settings state was `null` before the API returns. would that solve this problem?

Reply:
@jonahkagan That would break several things, actually. You need the structure there to be able to destructure from it. I think this right now is the least complex solution.

Reply:
you can check for null before you destructure. something like this:

```ts
const [auditSettings, updateAuditSettings] = useAuditSettings()

if (!auditSettings) return null

const {auditType} = auditSettings
```

the reason I suggest this is because it's the pattern that all the other API hooks use, and I think being consistent will make it easier to read/maintain the code

Reply:
@jonahkagan Okay, trying this is actually... really involved and changes a bunch of things across the board. So I'm merging this and introducing the refactor in a separate PR.